### PR TITLE
FIX: move hp request from /users to /token

### PR DESCRIPTION
**Pull request by lis2**
`hp` is a valid username and we should allow users to sign up with it.

### CODE REVIEW

File: spec/requests/users_controller_spec.rb [hunk 1]
```diff
@@ -1297,6 +1297,8 @@ def post_user(extra_params = {})
       before do
         UsersController.any_instance.stubs(:honeypot_value).returns("abc")
         UsersController.any_instance.stubs(:challenge_value).returns("efg")
```
Comment (tgxworld) [inline spec/requests/users_controller_spec.rb:1299]: Do we still need the stubs here?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -833,6 +836,14 @@ def add_noindex_header
 
   protected
 
+  def honeypot_value
```
Comment (tgxworld) [inline app/controllers/application_controller.rb:839]: Is there a reason we moved the methods to `ApplicationController` instead of `SessionController`?

**Review by tgxworld — APPROVED** [review]
Aside from some minor comments, the PR looks good.

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (lis2) [reply app/controllers/application_controller.rb:839]: Problem is that this is shared between SessionController and UsersController. In UsersController we check value before account is activated:
```
  def honeypot_or_challenge_fails?(params)
    return false if is_api?
    params[:password_confirmation] != honeypot_value ||
    params[:challenge] != challenge_value.try(:reverse)
  end
```

File: spec/requests/users_controller_spec.rb (hunk 1 above)
Comment (lis2) [reply spec/requests/users_controller_spec.rb:1299]: yes, SessionController is setting honeypot_value and challenge_value, and UsersController is checking that values are correct

**Pull request merged**